halotools/utils/rotations2d.py

```diff
@@ -0,0 +1,175 @@
+r"""
+A set of vector rotation utilites for manipulating 2-dimensional vectors
+"""
+
+from __future__ import (division, print_function, absolute_import,
+                        unicode_literals)
+import numpy as np
+from .vector_utilities import (elementwise_dot, elementwise_norm,
+                               normalized_vectors, angles_between_list_of_vectors)
+
+
+__all__=['rotation_matrices_from_angles',
+         'rotation_matrices_from_vectors',
+         'rotation_matrices_from_basis']
+__author__ = ['Duncan Campbell', 'Andrew Hearin']
+
+
+def rotation_matrices_from_angles(angles):
+    r"""
+    Calculate a collection of rotation matrices defined by
+    an input collection of rotation angles and rotation axes.
+
+    Parameters
+    ----------
+    angles : ndarray
+        Numpy array of shape (npts, ) storing a collection of rotation angles
+
+    Returns
+    -------
+    matrices : ndarray
+        Numpy array of shape (npts, 2, 2) storing a collection of rotation matrices
+
+    Examples
+    --------
+    >>> from halotools.utils.mcrotations import random_unit_vectors_2d
+    >>> npts = int(1e4)
+    >>> angles = np.random.uniform(-np.pi/2., np.pi/2., npts)
+    >>> rotation_matrices = rotation_matrices_from_angles(angles)
+
+    Notes
+    -----
+    The function `rotate_vector_collection` can be used to efficiently
+    apply the returned collection of matrices to a collection of 2d vectors
+    """
+
+    angles = np.atleast_1d(angles)
+    npts = len(angles)
+
+    sina = np.sin(angles)
+    cosa = np.cos(angles)
+
+    R = np.zeros((npts, 2, 2))
+    R[:, 0, 0] = cosa
+    R[:, 1, 1] = cosa
+
+    R[:, 0, 1] = -sina
+    R[:, 1, 0] = sina
+    
+    return R
+
+
+def rotation_matrices_from_vectors(v0, v1):
+    r"""
+    Calculate a collection of rotation matrices defined by the unique
+    transformation rotating v1 into v2.
```

Comment:
"defined by the unique transformation rotating v1 into v2 about their mutually perpendicular axis.

Reply:
Is your problem with the word "unique"?

Reply:
Right, yes, the rotation matrix that maps v0 into v1 is not unique, so as-written this docstring is incorrect. The modification I suggested actually just uses the same language used in some closely related function (I forget which).